core/common/src/main/scala/net/liftweb/common/Box.scala

```diff
@@ -785,34 +822,124 @@ sealed abstract class Box[+A] extends Product with Serializable{
    * Although this function is different for true collections, because `Box` is
    * really a collection of 1, the two functions are identical.
    */
-  final def collectFirst[B](pf: PartialFunction[A, B]): Box[B] = {
+  def collectFirst[B](pf: PartialFunction[A, B]): Box[B] = {
     collect(pf)
   }
 }
 
 /**
+ * A `Full` or `Empty` box (stands in trivially for `Option`).
+ */
+sealed trait PresenceBox[+A] extends Box[A] {
+  override def isA[B](clsOrg: Class[B]): PresenceBox[B] = Empty
+  override def asA[B](implicit m: Manifest[B]): PresenceBox[B] = Empty
+
+  def or[B >: A](alternative: => Full[B]): Full[B]
+  def or[B >: A](alternative: => PresenceBox[B])(implicit a: DummyImplicit): PresenceBox[B]
+  def or[B >: A](alternative: => TryBox[B])(implicit a: DummyImplicit, b: DummyImplicit): TryBox[B]
+  def or[B >: A, E](alternative: => ParamTryBox[B, E])(implicit a: DummyImplicit, b: DummyImplicit, c: DummyImplicit): ParamTryBox[B, E]
```

Comment:
I'm idly wondering if the presence of these `DummyImplicit`s might be a sign we've tried to be a bit too clever by incorporating this into the overall box structure instead of having them as a parallel data type with implicit conversions.